PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,5 +1,5 @@
-For an explanation of how to fill out the fields, please see the relevant section 
-in [PULL_REQUESTS.md](./PULL_REQUESTS.md)
```

Comment:
This linking already works?

Reply:
It works if you are viewing the file off master but when you open a pull request the link attempts to show a diff from `PULL_REQUEST.md`. :)

You can click the link above to see it redirects to https://github.com/envoyproxy/envoy/compare/PULL_REQUESTS.md?expand=1